pkg/controller/vitessshardreplication/init_restored_shard.go

```diff
@@ -231,7 +231,9 @@ func electInitialShardPrimary(ctx context.Context, keyspaceName, shardName strin
 		return nil, fmt.Errorf("lost topology lock; aborting: %v", err)
 	}
 	// Promote the candidate to primary.
-	_, err = wr.TabletManagerClient().PromoteReplica(ctx, candidatePrimary.tablet.Tablet)
+	// TODO (GuptaManan100): We are passing false for semiSync parameter since it is not being used currently.
+	// When we change this behaviour in Vitess, we should also revisit here.
```

Comment:
Can you document why we might change this behavior?

Reply:
Do you want me to add it in the code as more comments?

Reply:
link to a vitessio/vitess issue if we have one?